src/pages/en/guides/deploy/deno.md

```diff
@@ -6,47 +6,156 @@ layout: ~/layouts/DeployGuideLayout.astro
 
-    ```js
+    ```diff
+    // astro.config.mjs
     import { defineConfig } from 'astro/config';
-    import deno from "@astrojs/deno";
+    + import deno from '@astrojs/deno';
 
     export default defineConfig({
-      adapter: deno()
+    +   adapter: deno(),
     });
     ```
 
-2. Build your site by running `astro build`
+1. Update your `preview` script in `package.json` with the change below.
+
+    ```diff
+    // package.json
+    {
+      // ...
+      "scripts": {
+        "dev": "astro dev",
+        "start": "astro dev",
+        "build": "astro build",
+    -     "preview": "astro preview"
+    +     "preview": "deno run --allow-net --allow-read --allow-env ./dist/server/entry.mjs"
+      }
+    }
+    ```
+    
+    You can now use this command to preview your production Astro site locally with Deno.
+    
+    ```bash
+    npm run preview
+    ```
+
+## How to deploy
+
+You can deploy to Deno Deploy through Github Actions or using Deno Deploy’s CLI (command line interface).
+
+### Github Actions Deployment
+
+If your project is stored on GitHub, the [Deno Deploy website](https://dash.deno.com/) will guide you through setting up Github Actions to deploy your Astro site.
+
+1. Push your code to a public or private Github repository.
 
-3. Preview your local build with Deno
+1. Sign in on [Deno Deploy](https://dash.deno.com/) with your Github account, and click on [New Project](https://dash.deno.com).
 
-    To actually use Deno locally to preview your Astro site, you'll need to add some script changes.
+1. Select your repository, the branch you want to deploy from, and select **Github Action** mode. (Your Astro site requires a build step, and cannot use Automatic mode.)
+   
+1. In your Astro project, create a new file at `.github/workflows/deploy.yml` and paste in the YAML below. This is similar to the YAML given by Deno Deploy, with the additional steps needed for your Astro site.
 
-    Change the `package.json` scripts section to the following:
+    ```diff
+    name: Deploy
+    on: [push]
 
-    ```json
+    jobs:
+      deploy:
+        name: Deploy
+        runs-on: ubuntu-latest
+        permissions:
+          id-token: write # Needed for auth with Deno Deploy
+          contents: read # Needed to clone the repository
+
+        steps:
+          - name: Clone repository
+            uses: actions/checkout@v2
+
+    +       # Not using npm? Change `npm ci` to `yarn install` or `pnpm i`
+    +       - name: Install dependencies
+    +         run: npm ci
+    + 
+    +       # Not using npm? Change `npm run build` to `yarn build` or `pnpm run build`
+    +       - name: Build Astro
+    +         run: npm run build
+
+          - name: Upload to Deno Deploy
+            uses: denoland/deployctl@v1
+            with:
+    +           project: my-deno-project # TODO: replace with Deno Deploy project name
+    +           entrypoint: server/entry.mjs
+    +           root: dist
+    ```
```

Comment:
ah, I should switch this to YAML syntax (instead of diff), and rm the `+ ` stuff, so it's easier to copypasta.